### ISSUE TITLE
Add support for gnubin/stty on OSX (from Macports)

### ISSUE THREAD

**Issue by mgcrea** [amperka/ino#96]
```
$ ino upload
Searching for stty ... /opt/local/libexec/gnubin/stty
Searching for avrdude ... /Applications/Arduino.app/Contents/Resources/Java/hardware/tools/avr/bin/avrdude
Searching for avrdude.conf ... /Applications/Arduino.app/Contents/Resources/Java/hardware/tools/avr/etc/avrdude.conf
Guessing serial port ... /dev/tty.usbmodem1d1121
/opt/local/libexec/gnubin/stty: invalid argument ‘-f’
Try '/opt/local/libexec/gnubin/stty --help' for more information.
stty failed
```

**Comment by biffhero** [comment]
mgcrea, I edited /opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/ino/commands/upload.py on my machine, and changed one line.  My working line is now

file_switch = '-F' if platform.system() == 'Darwin' else '-F'

It used to have "-f" in that first part.  This allowed an upload to be successful from the command line.

**Comment by joshgoebel** [comment]
Is there a reason you're using stty from macports when it's a built-in on OSX already?

**Comment by mgcrea** [comment]
As a sysadmin, it's very nice to have scripts working the exact same way on both my local computer & remote servers (mostly debian ubuntu boxes). Using the GNU tools also brings up-to-date versions to OSX.

**Comment by joshgoebel** [comment]
Do you have some idea how to better auto-detect the proper param?  Obviously we're already trying to properly detect the proper flags.  Just GNU bin tools on OSX is an edge case that isn't handled.

**Comment by mgcrea** [comment]
@yyyc514 Could you use the absolute path for osx `/bin/stty` so that it gets compiled with the "known" tool no matter the config?

**Comment by joshgoebel** [comment]
You could, but that doesn't solve the actual case where a user wants to purposely use a newer GNU tools stack.

**Comment by biffhero** [comment]
Maybe this would work?

rob@host:~$ /bin/stty --help &> /dev/null 
rob@host:~$ echo $?
1
rob@host:~$ /opt/local/libexec/gnubin/stty --help &> /dev/null 
rob@host:~$ echo $?
0

**Comment by joshgoebel** [comment]
True, one could try and sniff the output (or the error code with --help)... not a bad suggestion.

**Comment by biffhero** [comment]
Yeah, I was hoping that both of them would have parseable output from --version, but no luck.

[61400]rob@host:~/Downloads$ /bin/stty --version
stty: illegal option -- --version
usage: stty [-a|-e|-g] [-f file] [options]
[65545]rob@host:~/Downloads$ echo $?
1
[69675]rob@host:~/Downloads$ /opt/local/libexec/gnubin/stty --version
stty (GNU coreutils) 8.22
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later http://gnu.org/licenses/gpl.html.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David MacKenzie.
[73822]rob@host:~/Downloads$ echo $?
0

**Comment by joshgoebel** [comment]
Honestly the existing code should be left as-is and a `:stty_options` configuration setting should be added for someone to pass custom flags since this is a edge edge case.
